src/Simulation/Simulators/QCTraceSimulator/QCTraceSimulator.cs

```diff
@@ -398,13 +398,19 @@ public string[] GetMetricStatisticNames(string metric)
         /// Creates a new instance of the simulator with the default 
         /// <see cref="QCTraceSimulatorConfiguration"/>.
         /// </summary>
-        public QCTraceSimulator() : this( new QCTraceSimulatorConfiguration() ) { }
+        public QCTraceSimulator(Type targetIntrinsicsType) : this(targetIntrinsicsType, new QCTraceSimulatorConfiguration() ) { }
+
+        /// <summary>
+        /// Creates a new instance of the simulator with configuration given by 
+        /// the <paramref name="config"/> parameter, using the target intrinsics from <paramref name="targetIntrinsicsType"/>.
+        /// </summary>
+        public QCTraceSimulator(Type targetIntrinsicsType, QCTraceSimulatorConfiguration config) : base(targetIntrinsicsType, config) { }
 
         /// <summary>
         /// Creates a new instance of the simulator with configuration given by 
         /// the <paramref name="config"/> parameter.
         /// </summary>
-        public QCTraceSimulator(QCTraceSimulatorConfiguration config) : base(config) { }
+        public QCTraceSimulator(QCTraceSimulatorConfiguration config) : base(FindTargetIntrinsicsType(), config) { }
```

Comment:
This would also go away if the C# generation created concrete classes instead of abstract classes, so I'm going to take a shot at removing it in favor of that approach.